### PR TITLE
add default nix formatter and its corresponding Github action

### PR DESCRIPTION
**Pull request by drupol**
This PR:

- [x] add a default formatter in the flake output, I used `alejandra`
- [x] add a new Github action workflow checking if running the formatter hasn't changed the files (the `--dry-run` hasn't been implemented yet, see https://github.com/NixOS/nix/issues/6918)

### CODE REVIEW

**Comment by drupol** [comment]
@jtojnar WDYT about this?

File: pkgs/phps.nix [hunk 1]
```diff
@@ -20,100 +13,94 @@ let
...
+          + attrs.preConfigure;
+      };
 
       # For passing pcre2 to php-packages.nix.
-      callPackage =
-        cpFn:
-        cpArgs:
-
+      callPackage = cpFn: cpArgs:
         prev.callPackage
```
Comment (jtojnar) [inline pkgs/phps.nix:62]: Here again the arguments are moved away from the function body’s level, making it harder to notice it is a function body when scanning the code.

File: pkgs/phps.nix [hunk 2]
```diff
@@ -20,100 +13,94 @@ let
...
-          cpFn
```
Comment (jtojnar) [inline pkgs/phps.nix:75]:
```
function
argument1
arg2
```

to me looks too much like a list of equal items so I have started indenting the arguments:

```
function
  argument1
  argument2
```

Will need to do more testing to see how it plays out in practice.

**Review by jtojnar — COMMENTED** [review]
I like this in principle and Alejandra feels closest to the ideal but there are still some cases where I feel like the formatting style hinders reading style where one is quickly scanning the code top-down (without moving much focus to the right).

I have wanted to play with it some more and argue for style changes where necessary upstream but have not found the time yet.

**Comment by drupol** [comment]
What we could do, is to use `nixpkgs-fmt` instead.

I saw many controversial things about `alejandra`.

IMHO, I like `alejandra`, but there are things I do not like in it as well, and I would not mind replacing it with something else.

File: pkgs/lib.nix [hunk 3]
```diff
@@ -1,42 +1,34 @@
 # SPDX-FileCopyrightText: 2023 Jan Tojnar <jtojnar@gmail.com>
 # SPDX-License-Identifier: MIT
+{lib}: {
+  /*
+  * Remove given lines from a string.
```
Comment (jtojnar) [inline pkgs/lib.nix:5]: Looks like doc comments are not supported either.

Comment (drupol) [reply pkgs/lib.nix:5]: Do you suggest another formatter ?

Comment (drupol) [reply pkgs/lib.nix:5]: There are:
- https://github.com/nix-community/nixpkgs-fmt
- https://github.com/serokell/nixfmt

Comment (jtojnar) [reply pkgs/lib.nix:5]: I think we can do without a formatter for now. I still plan to report the issues to Alejandra so that we can use it in future.

**Comment by drupol** [comment]
Closing the PR until further notice.

File: pkgs/lib.nix (hunk 3 above)
Comment (jtojnar) [reply pkgs/lib.nix:5]: Actually, I misremembered and documentation comments in Nixpkgs just use a single asterisk: https://github.com/NixOS/nixpkgs/blob/2abea38e0824edd57a3dcfe02f84e09a3a3734c5/lib/attrsets.nix#L888

**Pull request closed without merging**